replica/compaction_group.hh

```diff
@@ -38,8 +48,20 @@ using enable_backlog_tracker = bool_class<class enable_backlog_tracker_tag>;
 //          isolated from other groups.
 class compaction_group {
     table& _t;
+    // The compaction group views are the logical compaction groups, each having its own logical
+    // set of sstables. Even though they share the same instance of sstable_set, compaction will
+    // only see the sstables belonging to a particular view, with the help of the classifier.
+    // This way, we guarantee that sstables falling under different groups cannot be compacted
+    // together.
     class compaction_group_view;
-    std::unique_ptr<compaction_group_view> _compaction_group_view;
+    // This is held throughout group lifetime, in order to have compaction disabled on non-compacting views.
+    std::vector<compaction_manager::compaction_reenabler> _compaction_disabler_for_views;
```

Comment:
Can't we permanently disable compaction, without having to keep some RAII object alive?

Reply:
We could, but I felt like I could reuse something existing, rather than leaking a count here https://github.com/scylladb/scylladb/blob/master/compaction/compaction_state.hh#L31

What do you think? Maybe I can just set it to 1 in add_with_compaction_disabled() and that's it. And document that a compaction_group_view added through that interface in manager will never have compaciton enabled. The reason we want to register non-compacting view is for compatibility with table code iterating on all views and expecting them all to be registered in the manager.

Reply:
Ok, we can keep it as-is.